File: seed/ruby-sdk-v2/content-type/lib/seed/service/types/patch_proxy_request.rb
# frozen_string_literal: true

module Seed
  module Service
    module Types
      class PatchProxyRequest < Internal::Types::Model
        field :application, -> { String }, optional: false, nullable: true
        field :require_auth, -> { Internal::Types::Boolean }, optional: false, nullable: true
      end
    end
  end
end

File: seed/php-sdk/content-type/src/Service/Requests/PatchProxyRequest.php
<?php

namespace Seed\Service\Requests;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class PatchProxyRequest extends JsonSerializableType
{
    /**
     * @var ?string $application
     */
    #[JsonProperty('application')]
    public ?string $application;

    /**
     * @var ?bool $requireAuth
     */
    #[JsonProperty('require_auth')]
    public ?bool $requireAuth;

    /**
     * @param array{
     *   application?: ?string,
     *   requireAuth?: ?bool,
     * } $values
     */
    public function __construct(
        array $values = [],
    )
    {
        $this->application = $values['application'] ?? null;$this->requireAuth = $values['requireAuth'] ?? null;
    }
}
